src/util/saveArticles.ts

```diff
@@ -942,17 +942,15 @@ export function applyOtherTreatments(parsoidDoc: DominoElement, dump: Dump) {
 
   /* Remove empty paragraphs */
   if (!dump.opts.keepEmptyParagraphs) {
-    // Mobile view === details
-    // Desktop view === section
-    const sections: DominoElement[] = Array.from(parsoidDoc.querySelectorAll('details, section'))
-    for (const section of sections) {
-      if (
-        section.children.length ===
-        Array.from(section.children).filter((child: DominoElement) => {
-          return child.matches('summary')
-        }).length
-      ) {
-        DU.deleteNode(section)
+    const paragraphs: DominoElement[] = Array.from(parsoidDoc.querySelectorAll('p'))
+    const mediaTags = ['img', 'video', 'audio', 'embed', 'object', 'iframe', 'canvas', 'svg', 'picture', 'track', 'source']
+    for (const paragraph of paragraphs) {
+      const hasNoMediaContent = !mediaTags.some((tag) => paragraph.querySelector(tag))
+      // Check if no media content inside first
+      if (hasNoMediaContent) {
+        if (!paragraph.textContent || (paragraph.textContent && paragraph.textContent.trim().length === 0)) {
+          DU.deleteNode(paragraph)
+        }
```

Comment:
This does not behave like it should. Either it removes user visible empty parapgraph (`<h*>` DOM node or similar or it's wrong). Here it just removes `<p>` elements under certain conditions which is neither what is expected or what it was doing.

Reply:
Got it. I brought back the original implementation of this for now.